File: app/src/main/java/com/waveneuro/ui/base/BaseWebViewActivity.java
package com.waveneuro.ui.base;

import android.os.Bundle;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import androidx.annotation.Nullable;

public class BaseWebViewActivity extends BaseActivity {

    private WebView webView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    protected void setWebView(WebView webView) {
        this.webView = webView;
        this.webView.getSettings().setJavaScriptEnabled(true);
        this.webView.setWebViewClient(new WebViewClient());
        this.webView.setClickable(true);
        this.webView.setWebChromeClient(new WebChromeClient());
        this.webView.getSettings().setDomStorageEnabled(true);
        this.webView.getSettings().setSaveFormData(true);
        this.webView.getSettings().setAllowContentAccess(true);
    }

    protected void setUrl(String url) {
        this.webView.loadUrl(url);
    }

    protected void goBack() {
        if (webView.canGoBack()) {
            webView.goBack();
        } else {
            onBackPressed();
        }
    }
}

File: app/src/main/java/com/waveneuro/ui/base/activity/BaseWebViewActivity.kt
package com.waveneuro.ui.base.activity

import android.annotation.SuppressLint
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.viewbinding.ViewBinding

abstract class BaseWebViewActivity<B : ViewBinding> : BaseActivity<B>() {

    private var webView: WebView? = null

    @SuppressLint("SetJavaScriptEnabled")
    protected fun setWebView(webView: WebView?) {
        this.webView = webView
        this.webView?.webViewClient = WebViewClient()
        this.webView?.isClickable = true
        this.webView?.webChromeClient = WebChromeClient()
        this.webView?.settings?.domStorageEnabled = true
    }

    protected fun setUrl(url: String?) {
        webView?.loadUrl(url ?: "")
    }

    protected fun goBack() {
        if (webView?.canGoBack() == true) {
            webView?.goBack()
        } else {
            onBackPressed()
        }
    }

}